waku/v2/protocol/waku_store/waku_store.nim

```diff
@@ -459,24 +459,58 @@ proc query*(w: WakuStore, query: HistoryQuery, handler: QueryHandlerFunc) {.asyn
   waku_store_messages.set(response.value.response.messages.len.int64, labelValues = ["retrieved"])
   handler(response.value.response)
 
+proc queryFrom*(w: WakuStore, query: HistoryQuery, handler: QueryHandlerFunc, peer: PeerInfo): Future[bool] {.async.} =
+  # sends the query to the given peer
+  # returns true if no error occurs, otherwise false
+  let connOpt = await w.peerManager.dialPeer(peer, WakuStoreCodec)
+  echo "here"
+
+  if connOpt.isNone():
+    error "failed to connect to remote peer"
+    waku_store_errors.inc(labelValues = [dialFailure])
+    return false
+
+  await connOpt.get().writeLP(HistoryRPC(requestId: generateRequestId(w.rng),
+      query: query).encode().buffer)
+
+  var message = await connOpt.get().readLp(64*1024)
+  let response = HistoryRPC.init(message)
+
+  if response.isErr:
```

Comment:
I wonder if we want to make this into a Result type, so we can propagate error to caller of function? Instead of bool.

Reply:
Done!